packages/amplify-codegen/src/utils/validateDartSDK.js

```diff
@@ -0,0 +1,26 @@
+const yaml = require('js-yaml');
+const path = require('path');
+const fs = require('fs-extra');
+const semver = require('semver');
+
+const pubspecFile = 'pubspec.yaml';
+
+function validateDartSDK(context, projectRoot) {
+  try {
+    const config = yaml.load(fs.readFileSync(path.join(projectRoot, pubspecFile), 'utf8'));
+    const version = semver.minVersion(config.environment.sdk);
+    if (semver.satisfies(version, '>= 2.12.0')) {
+      context.print.warning(
+        '\nDetect dart SDK satisfies the minimum version of 2.12.0. Null safety feature will be applied in new projects while old projects need to switch the feature flag value manually.',
+      );
+      return true;
+    }
+    context.print.warning('\nDetect dart SDK does not satisfy the minimum version of 2.12.0. Null safety feature will be disabled.');
+    return false;
+  } catch (e) {
+    context.print.warning('\nCannot find dart sdk version. Null safety feature will be disabled.');
```

Comment:
So I believe that if the pubspec.yaml cannot be found or the version cannot be determined, we should make the codegen version determination using the feature flag.  I'm guessing that this warning should be changed to simply '\nCannot find dart sdk version' and the function can return true?